File: src/text_utils.py
import PyPDF2
from colorama import Fore


def extract_text_from_pdf(pdf_path):
    """Extract text from PDF."""

    with open(pdf_path, "rb") as file:
        reader = PyPDF2.PdfReader(file)
        text = ""
        for page in reader.pages:
            text += page.extract_text()

    return text


def split_texts(text: str) -> tuple:
    """Split the text into header and food service items."""

    lines = text.split("\n")
    split_index = None
    for i, line in enumerate(lines):
        if line.strip() == "Food/Service Item":
            split_index = i
            break
    if split_index is not None:
        header_text = "\n".join(lines[: split_index + 1])
        remaining_text = "\n".join(lines[split_index + 1 :])
    else:
        header_text = text
        remaining_text = ""

    return (header_text, remaining_text)


def print_stars():
    print_yellow("**********************************************")


def print_red(text):
    print(Fore.RED + text)


def print_green(text):
    print(Fore.GREEN + text)


def print_yellow(text):
    print(Fore.YELLOW + text)


def print_cyan(text):
    print(Fore.CYAN + text)


def print_stats(text):
    """Pretty prints statistics as yellow text until the colon."""
    split_text = text.split(":")
    print_yellow(split_text[0] + ":" + Fore.RESET + split_text[1])


def print_header(text):
    """Prints a yellow star and the rest in white."""
    print_stars()
    print_yellow("*")
    print_yellow("* " + Fore.RESET + text)
    print_yellow("*\n")

File: src/utils/text_utils.go
package utils

import (
	"bytes"
	"fmt"
	"strings"

	"github.com/fatih/color"
	"github.com/ledongthuc/pdf"
)

// extractTextFromPDF extracts text from a PDF file.
func ExtractTextFromPDF(pdfPath string) (string, error) {
	f, r, err := pdf.Open(pdfPath)
	// remember close file
	defer f.Close()
	if err != nil {
		return "", err
	}
	var buf bytes.Buffer
	b, err := r.GetPlainText()
	if err != nil {
		return "", err
	}
	buf.ReadFrom(b)
	return buf.String(), nil
}

// splitTexts splits the text into header and food service items.
func SplitTexts(text string) (headerText, remainingText string) {
	lines := strings.Split(text, "\n")
	splitIndex := -1
	for i, line := range lines {
		if strings.TrimSpace(line) == "Food/Service Item" {
			splitIndex = i
			break
		}
	}
	if splitIndex != -1 {
		headerText = strings.Join(lines[:splitIndex+1], "\n")
		remainingText = strings.Join(lines[splitIndex+1:], "\n")
	} else {
		headerText = text
		remainingText = ""
	}
	return headerText, remainingText
}

// printStars prints a line of stars in yellow.
func PrintStars() {
	PrintYellow("**********************************************")
}

// printRed prints text in red.
func PrintRed(text string) {
	color.Red(text)
}

// printGreen prints text in green.
func PrintGreen(text string) {
	color.Green(text)
}

// printYellow prints text in yellow.
func PrintYellow(text string) {
	color.Yellow(text)
}

// printCyan prints text in cyan.
func PrintCyan(text string) {
	color.Cyan(text)
}

// printStats pretty prints statistics as yellow text until the colon.
func PrintStats(text string) {
	splitText := strings.Split(text, ":")
	PrintYellow(splitText[0] + ":" + splitText[1])
}

// printHeader prints a yellow star and the rest in white.
func PrintHeader(text string) {
	PrintStars()
	PrintYellow("*")
	PrintYellow("* " + text)
	PrintYellow("*")
	fmt.Println()
}
